pkg/actions/addon/create.go

```diff
@@ -40,6 +40,14 @@ func (a *Manager) Create(addon *api.Addon) error {
 	logger.Debug("addon: %v", addon)
 	namespace, serviceAccount := a.getKnownServiceAccountLocation(addon)
 
+	if addon.Tags != nil && len(addon.Tags) > 0 {
+		logger.Info("using provided Tags %q", addon.Tags)
+		awsTags := map[string]*string{}
+		for k, v := range addon.Tags {
+			awsTags[k] = aws.String(v)
+		}
+		createAddonInput.Tags = awsTags
```

Comment:
The AWS SDK has a handy utility for this:

```suggestion
		createAddonInput.Tags = aws.StringMap(addon.Tags)
```